app/scripts/metamask-controller.js

```diff
@@ -1982,6 +2060,22 @@ export default class MetamaskController extends EventEmitter {
         ///: END:ONLY_INCLUDE_IN
         ///: BEGIN:ONLY_INCLUDE_IN(keyring-snaps)
         getSnapKeyring: this.getSnapKeyring.bind(this),
+        requestUserApproval:
```

Comment:
Are these hooks still needed if we're adding the logic above, or are we adding similar flows in the extension and the `rpc-methods` package?

Reply:
Good catch. These are no longer used and I'll remove them.